packages/core-utils/src/itinerary.ts

```diff
@@ -564,18 +564,19 @@ export function getDisplayedStopId(placeOrStop: Place | Stop): string {
  */
 export function getLegCost(
   leg: Leg,
-  mediumId: string,
-  riderCategoryId: string
+  mediumId: string | null,
+  riderCategoryId: string | null
 ): { price?: Money; transferAmount?: Money | undefined } {
   if (!leg.fareProducts) return { price: undefined };
   const relevantFareProducts = leg.fareProducts.filter(({ product }) => {
     return (
-      product.riderCategory.id === riderCategoryId &&
-      product.medium.id === mediumId
+      (product.riderCategory === null ? null : product.riderCategory.id) ===
```

Comment:
Can this simply be written as `product.riderCategory?.id`? Same for `product.medium`.

Reply:
We do this to handle null differently from undefined

Reply:
Thanks for the clarification. Add what you said in a comment above that code, please.